gadgets/trace_open/test/unit/trace_open_test.go

```diff
@@ -0,0 +1,297 @@
+
+package tests
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/unix"
+
+	gadgetapi "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf"
+)
+
+type ExpectedTraceOpenEvent struct {
+	Comm  string `json:"comm"`
+	Pid   int    `json:"pid"`
+	Tid   int    `json:"tid"`
+	Uid   uint32 `json:"uid"`
+	Gid   uint32 `json:"gid"`
+	Fd    uint32 `json:"fd"`
+	FName string `json:"fname"`
+}
+
+type testDef struct {
+	runnerConfig  *utilstest.RunnerConfig
+	mntsFilterMap func(info *utilstest.RunnerInfo) *ebpf.Map
+	generateEvent func(t *testing.T) int
+	validateEvent func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedTraceOpenEvent) error
+}
+
+func TestTraceOpenGadget(t *testing.T) {
+	utilstest.RequireRoot(t)
+	testCases := map[string]testDef{
+		"captures_all_events_with_no_filters_configured": {
+			runnerConfig:  &utilstest.RunnerConfig{},
+			generateEvent: generateEvent,
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedTraceOpenEvent) error {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, fd int) *ExpectedTraceOpenEvent {
+					return &ExpectedTraceOpenEvent{
+						Comm:  info.Comm,
+						Pid:   info.Pid,
+						Tid:   info.Tid,
+						Uid:   uint32(info.Uid),
+						Gid:   uint32(info.Gid),
+						Fd:    uint32(fd),
+						FName: "/dev/null",
+					}
+				})(t, info, fd, events)
+				return nil
+			},
+		},
+		"captures_no_events_with_no_matching_filter": {
+			runnerConfig: &utilstest.RunnerConfig{},
+			mntsFilterMap: func(info *utilstest.RunnerInfo) *ebpf.Map {
+				return utilstest.CreateMntNsFilterMap(t, 0)
+			},
+			generateEvent: generateEvent,
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedTraceOpenEvent) error {
+				utilstest.ExpectNoEvent(t, info, fd, events)
+				return nil
+			},
+		},
+		"captures_events_with_matching_filter": {
+			runnerConfig: &utilstest.RunnerConfig{},
+			mntsFilterMap: func(info *utilstest.RunnerInfo) *ebpf.Map {
+				return utilstest.CreateMntNsFilterMap(t, info.MountNsID)
+			},
+			generateEvent: generateEvent,
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedTraceOpenEvent) error {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, fd int) *ExpectedTraceOpenEvent {
+					return &ExpectedTraceOpenEvent{
+						Comm:  info.Comm,
+						Pid:   info.Pid,
+						Tid:   info.Tid,
+						Uid:   uint32(info.Uid),
+						Gid:   uint32(info.Gid),
+						Fd:    uint32(fd),
+						FName: "/dev/null",
+					}
+				})(t, info, fd, events)
+				return nil
+			},
+		},
+		"test_flags_and_mode": {
```

Comment:
Flags nor mode are being tested. Are they missing from the event struct definition and so on?